Enqueue email alerts with Sidekiq after a successful publish

tuzz opened the pull request:
https://trello.com/c/gwGF7DA8/666-enqueue-travel-advice-email-alert-with-sidekiq

The commits are fairly descriptive if you need more context of specific changes.

boffbowsh commented:
Overall, 👍  but can you squash some commits please? `check_email_alert_happens_last` is introduced and replaced in this pull request

danielroseman commented on app/workers/worker_error.rb:
```diff
@@ -0,0 +1,31 @@
+  end
+
+  def queue_size
+    queue = Sidekiq::Queue.all.first
```
While it's useful to have this (and the retry_set size) in the error log, is there a possibility that since it's calling out to an API it could fail? What would happen to the error message if it did?

tuzz replied:
Good spot. There is that chance and that would cause the error message generation to exception and we'd get something cryptic in Errbit.

I will wrap these methods in a `rescue` to mitigate this risk.

dougdroper replied:
Is this somehow different to `queue = Sidekiq::Queue.first`?

tuzz replied:
`Sidekiq::Queue` doesn't respond to `.first`. It only provides an `.all` for whatever reason.

dougdroper replied:
Ah well :( :+1:

tuzz commented:
@boffbowsh I have cleaned up the commit messages. I made it so that batch check is atomically introduced in the latter commit, which had a descriptive commit message.

@danielroseman I have added a blanket raise to the methods that add Sidekiq information to the error message. This is currently a blanket raise, which might be overly zealous, but seemed safest.

Please could you both take another look. Thanks.

boffbowsh commented:
👍 from me

tuzz commented:
I think @danielroseman will be busy for the rest of the day. Please could @elliotcm or @dougdroper take a look?

dougdroper commented on app/notifiers/publishing_api_notifier.rb:
```diff
@@ -42,4 +51,25 @@ def enqueue
+    edition.state == "published" && !edition.minor_update
+  end
+
+  def check_tasks_are_ordered_sensibly
```
this method feels more like `validate_tasks_order` but not a big deal

tuzz replied:
👍  updated

dougdroper commented on app/notifiers/publishing_api_notifier.rb:
```diff
@@ -31,8 +31,17 @@ def publish_index
   end
 
+  def send_alert(edition)
+    if send_alert?(edition)
```
I think a prefer guards rather. `return if send_alert?(edition)` but also not a big deal.

tuzz replied:
👍  updated

tuzz commented:
@dougdroper said he was happy for this to be merged after rebase, which is now done.
